apps/web/src/components/landing/Footer.tsx

```diff
@@ -1,23 +1,170 @@
 "use client";
 
-import { type FunctionComponent, useState } from "react";
+import { useEffect, useState } from "react";
 import Image from "next/image";
 import Link from "next/link";
-import { Instagram, Facebook, Twitter, Github, Disc } from "lucide-react";
-import Discord from "../../../public/img/landing/discord_icon.svg";
+import { Instagram, Facebook, Twitter, Github } from "lucide-react";
+import {
+  DropdownMenu,
+  DropdownMenuTrigger,
+  DropdownMenuItem,
+  DropdownMenuContent,
+} from "../shadcn/ui/dropdown-menu";
 
 interface Props {
-	className?: string;
+  className?: string;
 }
 
+interface LinkData {
+  name: string;
+  link: string;
+}
+
+function linkItem(name: string, link: string): LinkData {
+  return { name, link };
+}
+
+// Maybe source this from the configuration file so that other hackathons can define their links in a more declarative way.
+const dropdownItemMap = {
+  Resources: [
+    linkItem("Register", "/register"),
+    linkItem("FAQ", "/faq"),
+    linkItem("Code of Conduct", "/conduct"),
+    linkItem("Contact Us", "/contact"),
+    linkItem("ACM-W", "https://acmutsa.org/suborg_acmw"),
+    linkItem("ACM UTSA", "https://acmutsa.org/"),
+  ],
+  Links: [
+    linkItem("Open Source", "https://github.com/acmutsa/CodeQuantum2023"),
+  ],
+  "Other Hackathons": [
+    linkItem("CodeQuantum", "https://cqhacks.org/"),
+    linkItem("RowdyDatathon", "https://www.rowdydatathon.org/"),
+    linkItem("TAMUHack", "https://tamuhack.com/"),
+    linkItem("WEHack", "https://wehackutd.com/"),
+    linkItem("HackUTD", "https://hackutd.co/"),
+    linkItem("HackTX", "https://hacktx.com/"),
+    linkItem("HackUNT", "https://unthackathon.com/#/"),
+    linkItem("HackUTA", "https://hackuta.org/"),
+    linkItem("Hacklahoma", "https://hacklahoma.org/"),
+  ],
+};
+
 export default function Footer() {
-	const [showResources, setShowResources] = useState(false);
-	const [showLinks, setShowLinks] = useState(false);
-	const [showHackathons, setShowHackathons] = useState(false);
+  const LinkSection = ({
+    title,
+    data,
+  }: {
+    title: string;
+    data: LinkData[];
+  }) => {
+    const [small, setSmall] = useState(true);
+
+    useEffect(() => {
```

Comment:
non-blocking: Part of why we use React is for their layer of abstraction over the DOM itself. This could also be achieved with using media queries directly in the `className` object. Also, it looks like this is the only use of state since you removed the show dropdowns so if we use media queries, we could get some more of this component in SSR to my knowledge.

Reply:
Should be amended by c2139a0